objects/src/mock/notes.rs

```diff
@@ -89,22 +89,20 @@ pub fn mock_notes(
     // create note 1 script
     let note_1_script_src = format!(
         "\
-        use.miden::sat::tx
-
         begin
             # create note 0
             push.{created_note_0_recipient}
             push.{created_note_0_tag}
             push.{created_note_0_asset}
-            exec.tx::create_note
-            drop 
+            call.0x33e6e544bce56ad3bd235d9806e23910a4f03371e7e0a8549d730d0ee61266d8
+            drop dropw dropw 
```

Comment:
I started out with this logic defined in a procedure at the top of this source file using the context import:

```
use.context::account_*

proc.create_note_wrapper
      call.account_*::create_note
      drop dropw dropw
end
```

This was also defined at the top of the source file in `note_2_script_src`.  This led to a strange bug which I suspect is related to compiler cache collision or some other form of unexpected behaviour which I haven't had time to investigate.  Migrating to the hex specifier of the account procedure fixed things.

Reply:
I think using hex-based calling convention is the right approach here and I've created https://github.com/0xPolygonMiden/miden-vm/issues/1071 to improve support for this. But it is a little concerning that there was this odd behavior in the first place. Do you think it would be possible to create a minimal reproducible example of this in Miden VM? If we have such an example, I should be able to debug this relatively quickly.

Reply:
Sure thing, I will try and recreate an example.

Reply:
Here is an example of the failure case for the transaction in which we use different names for the procedures at the top of each note script:

https://github.com/0xPolygonMiden/miden-base/compare/frisitano-create-note-auth...frisitano-assembler-error

This actually compiles and runs but fails an assertion error which I wasn't able to track down.

Here is a problem related to procedure name collisions in different scripts when compiled using the same `AssemblyContext`:

https://github.com/0xPolygonMiden/miden-vm/compare/next...frisitano-assembler-proc-collisions

Reply:
When I inline the `call` operation it works fine so maybe it's something to do with the way that call operations are inlined when invoked via procedures?

This works fine:
https://github.com/0xPolygonMiden/miden-base/compare/frisitano-create-note-auth...frisitano-assembler-bug-2